pilot/pkg/serviceregistry/kube/controller/ambient/waypoints.go

```diff
@@ -378,24 +380,50 @@ func (w *Waypoint) GetAddress() *workloadapi.GatewayAddress {
 	return w.Address
 }
 
-func makeAllowedRoutes(gateway *v1beta1.Gateway) WaypointSelector {
+// makeAllowedRoutes returns a WaypointSelector that matches the listener with the given binding
+// if we don't have a binding we use the default HBONE listener
+// if we have a binding we use the protocol and port defined in the binding
+func makeAllowedRoutes(gateway *v1beta1.Gateway, binding *InboundBinding) WaypointSelector {
+	matchProtocol := sets.New[gatewayv1.ProtocolType]()
+	matchPort := gatewayv1.PortNumber(15008)
+	// if we have a binding we should use it's protocol and port
+	if binding != nil {
+		// binding.Port to PortNumber is a conversion from uint32 to int32
+		matchPort = gatewayv1.PortNumber(binding.Port)
+		switch binding.Protocol {
+		case workloadapi.ApplicationTunnel_PROXY:
+			matchProtocol.Insert(gatewayv1.ProtocolType(constants.WaypointSandwichListenerProxyProtocol))
+			matchProtocol.Insert(gatewayv1.ProtocolType("PROXY"))
```

Comment:
1. `NONE` probably doesn't have an actual use outside being the zero-value that means there is no inbound binding.  NONE with a non-zero port should probably be interpreted as "the waypoint listens on this port, sandwiching ztunnel still listens on 15008". 

2. "PROXY" isn't really a valid protocol for Listener. The [spec](https://gateway-api.sigs.k8s.io/reference/spec/#protocoltype:~:text=Implementation%2Dspecific%20conformance.-,Valid%20values%20include,-%3A) says to use a `hostname.com/` prefix for impl specific stuff.

I think if `binding != null` we just try to match the port # before falling back to the old logic for HBONE